File: frontend/src/types/task.ts
export type TaskStatus = 'pendiente' | 'completada';

export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
}

export interface TaskHistory {
  id: string;
  status: TaskStatus;
  date: string;
}

export interface CreateTaskPayload {
  title: string;
  description?: string;
}

export interface UpdateTaskPayload {
  title: string;
  description?: string;
}

File: backend/tasks-creator/domain/model/src/main/java/com/taskscreator/model/Task.java
package com.taskscreator.model;

import lombok.Builder;
import lombok.Getter;

import java.time.OffsetDateTime;
import java.util.UUID;

@Getter
@Builder
public class Task {
    private UUID id;
    private String title;
    private String description;
    private TaskStatus status;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
